pkg/karmadactl/get.go

```diff
@@ -0,0 +1,559 @@
+	}
+
+	if err := g.getRbInKarmada(karmadaclient); err != nil {
+		return nil, err
+	}
+
+	if len(g.Clusters) <= 0 {
+		for c := range clusterInfos {
+			g.Clusters = append(g.Clusters, c)
+		}
+	}
+	return karmadaclient, nil
+}
+
+func getFactory(clusterName string, clusterInfos map[string]*ClusterInfo) cmdutil.Factory {
+	kubeConfigFlags := NewConfigFlags(true).WithDeprecatedPasswordFlag()
+	// Build member cluster kubeConfigFlags
+	kubeConfigFlags.BearerToken = stringptr(clusterInfos[clusterName].BearerToken)
+	kubeConfigFlags.APIServer = stringptr(clusterInfos[clusterName].APIEndpoint)
+	kubeConfigFlags.CaBundle = stringptr(clusterInfos[clusterName].CAData)
+	matchVersionKubeConfigFlags := cmdutil.NewMatchVersionFlags(kubeConfigFlags)
+	return cmdutil.NewFactory(matchVersionKubeConfigFlags)
+}
+
+func (g *CommandGetOptions) transformRequests(req *rest.Request) {
+	// We need full objects if printing with openapi columns
+	if g.PrintWithOpenAPICols {
+		return
+	}
+	if !g.ServerPrint || !g.IsHumanReadablePrinter {
+		return
+	}
+
+	req.SetHeader("Accept", strings.Join([]string{
+		fmt.Sprintf("application/json;as=Table;v=%s;g=%s", metav1.SchemeGroupVersion.Version, metav1.GroupName),
+		fmt.Sprintf("application/json;as=Table;v=%s;g=%s", metav1beta1.SchemeGroupVersion.Version, metav1beta1.GroupName),
+		"application/json",
+	}, ","))
+}
+
+func (g *CommandGetOptions) getRbInKarmada(config *rest.Config) error {
+	objectGVR := schema.GroupVersionResource{
+		Group:    "work.karmada.io",
+		Version:  "v1alpha1",
+		Resource: "resourcebindings",
+	}
+	ls := fmt.Sprintf("%s=%s", rbLabelNSKey, g.Namespace)
+	clusterDyClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	clusterList, err := clusterDyClient.Resource(objectGVR).List(context.TODO(), metav1.ListOptions{LabelSelector: ls})
+	if err != nil {
+		return err
+	}
+	for idx := range clusterList.Items {
+		rbKey := clusterList.Items[idx].GetName()
+		val, found, err := unstructured.NestedSlice(clusterList.Items[idx].Object, "status", "aggregatedStatus")
+		if !found || err != nil {
+			return errors.Wrap(err, fmt.Sprintf("Func getRbInKarmada not get aggregatedStatus in %s status", rbKey))
+		}
+		for i := range val {
+			if v, ok := val[i].(map[string]interface{}); ok {
+				if v["applied"] != nil && v["clusterName"] != nil {
+					newRbKey := fmt.Sprintf("%s-%s", v["clusterName"], rbKey)
+					RBInfo[newRbKey] = &OtherPrint{
+						Applied: v["applied"],
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// getSecretTokenInKarmada get token ca in karmada cluster
+func (g *CommandGetOptions) getSecretTokenInKarmada(client *rest.Config, name string, clusterInfos map[string]*ClusterInfo) error {
+	clusterClient, err := kubernetes.NewForConfig(client)
+	if err != nil {
+		return err
+	}
+	secret, err := clusterClient.CoreV1().Secrets("karmada-cluster").Get(context.TODO(), name, metav1.GetOptions{})
```

Comment:
If cluster mode is `pull`, we don't have a secret in `karmada-cluster`. Maybe we could fix it in next PR.

Reply:
Yes, now I can't get the member cluster for pull mode,  next PR let's think of a solution.

Reply:
I run this command on the environment created by `hack/local-up-karmada.sh`:
```
# ./karmadactl get deployment
error: Method getSecretTokenInKarmada get Secret info in karmada failed, err is: secrets "member3" not found
```
So, I suggest ignore `Pull` mode clusters temporarily, for people to try it without suprise.